src/celeritas/global/alongstep/AlongStep.hh

```diff
@@ -84,6 +84,13 @@ inline CELER_FUNCTION void along_step(MH&&                 msc,
             local.geo_step          = p.distance;
             local.step_limit.action = track.boundary_action();
         }
+        else if (p.distance < local.geo_step)
+        {
+            // Some other internal non-boundary geometry limit has been reached
+            // (e.g. too many substeps)
+            local.geo_step          = p.distance;
+            local.step_limit.action = track.propagation_limit_action();
+        }
```

Comment:
in `UrbanMscScatter` we check if a track is geometry limited by checking if it's on a boundary -- should we also be checking for this case now?

Reply:
Ooh, good call. Yes, we probably need to do that for correctness. I don't think this should happen often enough to make a difference in results, but I'm not sure what the consequences would be if we don't.